Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -56,16 +56,31 @@ bool LogFileReader::readAll(const QString& fileName)
  * Need to implement error checking for the correct number of values and
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
-{
+{  
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if(sections.size() != COLUMNS)
+        return false;
+
     QString timeString = sections.at(0);
+
     batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
+    bool validDoubleVoltage;
 
-    batteryData.current = sections.at(2).toDouble();
+    bool validDoubleCurrent;
 
-    return true;
+    batteryData.voltage = sections.at(1).toDouble(&validDoubleVoltage);
+
+    batteryData.current = sections.at(2).toDouble(&validDoubleCurrent);
```

Comment:
Instead of spacing lines out 1 line each, I would suggest grouping similar portions of code together, ie. group the creating of the voltage & current together since they are so similar.